Automattic/jetpack
# Display location-aware snackbar upon loading the site editor

Fixes https://github.com/Automattic/wp-calypso/issues/71274

## Proposed changes:
Adds a new Gutenberg extension that only runs in the site editor. The new extension shows a snackbar after entering the edit mode to better inform users what's being edited.

Template | Template part
--- | ---
<img width="1280" alt="Screenshot 2023-02-07 at 16 55 07" src="https://user-images.githubusercontent.com/1233880/217297323-881168c3-b922-4fdc-b8a1-2ac2121120b7.png"> | <img width="1279" alt="Screenshot 2023-02-07 at 16 55 58" src="https://user-images.githubusercontent.com/1233880/217297342-eefa2008-7d2c-4311-ac65-1273b3bff0e7.png">

### Other information:


## Jetpack product discussion
N/A

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
- Apply these changes to a test site:
  - Jetpack site: [Spin up a JN site running this branch](https://jurassic.ninja/create/?jetpack-beta&branches.jetpack=add/site-editor-snackbar-load) and activate the latest version of Gutenberg.
  - WP.com Simple site: Run `bin/jetpack-downloader test jetpack add/site-editor-snackbar-load` on your sandbox.
  - WP.com Atomic site: Install Jetpack Beta and activate this branch.
- Go to Appearance > Editor.
- Make sure no snackbars are displayed when you're on the view/readonly mode.
  - Note that this mode is only available in recent versions of Gutenberg.
  - Also note that this mode is skipped in WP.com when accessing the site editor via Calypso.
- Select a template or template part and enter the edit mode.
- Make sure a snackbar is displayed indicating the title and type of the entity being edited.
- Go to Posts > Add New.
- Make sure no snackbars are displayed and that there aren't any JS errors produced by these changes.
- Deactivate the Gutenberg plugin (note that this is only possible on a Jetpack / Atomic site).
- Go to Appearance > Editor.
- Make sure no snackbars are displayed and that there aren't any JS errors produced by these changes.
